src/Sentry.Unity/WebGL/SentryWebGL.cs

```diff
@@ -0,0 +1,163 @@
+            //    "An abnormal situation has occurred: the PlayerLoop internal function has been called recursively.
+            //     Please contact Customer Support with a sample project so that we can reproduce the problem and troubleshoot it."
+            // Maybe we could write a file when this error occurs and recognize it on the next start. Like unity-native.
+            options.CrashedLastRun = () => false;
+        }
+    }
+
+    internal class WebBackgroundWorker : IBackgroundWorker
+    {
+        private readonly SentryMonoBehaviour _behaviour;
+        private readonly UnityWebRequestTransport _transport;
+
+        public WebBackgroundWorker(SentryUnityOptions options, SentryMonoBehaviour behaviour)
+        {
+            _behaviour = behaviour;
+            _transport = new UnityWebRequestTransport(options);
+        }
+
+        public bool EnqueueEnvelope(Envelope envelope)
+        {
+            _ = _behaviour.StartCoroutine(_transport.SendEnvelopeAsync(envelope));
+            return true;
+        }
+
+        public Task FlushAsync(TimeSpan timeout) => Task.CompletedTask;
+
+        public int QueuedItems { get; }
+    }
+
+    internal class UnityWebRequestTransport : HttpTransportBase
+    {
+        private readonly SentryUnityOptions _options;
+
+        public UnityWebRequestTransport(SentryUnityOptions options)
+            : base(options)
+        {
+            _options = options;
+        }
+
+        // adapted HttpTransport.SendEnvelopeAsync()
+        internal IEnumerator SendEnvelopeAsync(Envelope envelope)
+        {
+            using var processedEnvelope = ProcessEnvelope(envelope);
+            if (processedEnvelope.Items.Count > 0)
+            {
+                // Send envelope to ingress
+                var httpRequest = CreateRequest(processedEnvelope);
+                var www = CreateWebRequest(httpRequest, processedEnvelope);
+                yield return www.SendWebRequest();
+
+                var response = GetResponse(www);
+                if (response != null)
+                {
+                    HandleResponse(response, processedEnvelope);
+                }
+            }
+        }
+
+        private UnityWebRequest CreateWebRequest(HttpRequestMessage message, Envelope envelope)
+        {
+            // Note: In order to use the synchronous Envelope.Serialize() we ignore the `message.Content`
+            // which is an `EnvelopeHttpContent` instance and use the actual envelope it wraps.
+            var stream = new MemoryStream();
+            try
+            {
+                envelope.Serialize(stream, _options.DiagnosticLogger);
+                stream.Flush();
+            }
+            catch (Exception e)
+            {
+                _options.DiagnosticLogger?.LogError("Failed to serialize Envelope into the network stream", e);
+                throw;
+            }
```

Comment:
@mattjohnsonpint do you think this is acceptable or do you want to expose sentry-dotnet's `EnvelopeHttpContent` & add a synchronous `SerializeToStream` method?

Reply:
Thanks for pointing that out.  I made a couple more changes.  You should now just be able to do the following instead:

```csharp
using var stream = ReadStreamFromHttpContent(message.Content);
```

Reply:
thx